From 8f87339354585949ba8033196b1e8b57623f5a52 Mon Sep 17 00:00:00 2001
From: Henry Gebhardt <hsggebhardt@googlemail.com>
Date: Wed, 29 Feb 2012 14:14:33 +0100
Subject: [PATCH 146/219] Fix empty systray icon in some panels on battery
 removal and addition

Unreferencing a GtkStatusIcon is not enough to get rid of it entirely.

This fixes these bugs:

    https://bugzilla.xfce.org/show_bug.cgi?id=7603
    http://bugzilla.xfce.org/show_bug.cgi?id=8424
    https://bugs.launchpad.net/ubuntu/+source/lxpanel/+bug/846878
    https://bugzilla.redhat.com/show_bug.cgi?id=765726

Signed-off-by: Eric Koegel <eric.koegel@gmail.com>
---
 src/xfpm-battery.c | 2 ++
 src/xfpm-power.c   | 1 +
 2 files changed, 3 insertions(+)

diff --git a/src/xfpm-battery.c b/src/xfpm-battery.c
index b3690d2..2e0e504 100644
--- a/src/xfpm-battery.c
+++ b/src/xfpm-battery.c
@@ -788,6 +788,8 @@ xfpm_battery_finalize (GObject *object)
     g_object_unref (battery->priv->notify);
     g_object_unref (battery->priv->button);
 
+    gtk_status_icon_set_visible(GTK_STATUS_ICON(battery), FALSE);
+
     G_OBJECT_CLASS (xfpm_battery_parent_class)->finalize (object);
 }
 
diff --git a/src/xfpm-power.c b/src/xfpm-power.c
index a53753f..febac1d 100644
--- a/src/xfpm-power.c
+++ b/src/xfpm-power.c
@@ -1137,6 +1137,7 @@ xfpm_power_hide_adapter_icon (XfpmPower *power)
 
     if ( power->priv->adapter_icon )
     {
+        gtk_status_icon_set_visible (power->priv->adapter_icon, FALSE);
         g_object_unref (power->priv->adapter_icon);
         power->priv->adapter_icon = NULL;
     }
-- 
1.9.3

